src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * Contains the main logic for responses by Duke.
+ */
+public class Parser {
+    /**
+     * Executes operations on the task list based on given input and
+     * returns the response by the program.
+     *
+     * @param input Input from user.
+     * @param tasks List of tasks.
+     * @return Program response as a String.
+     * @throws DukeException If input is not recognised.
+     */
+    public static String process(String input, TaskList tasks) throws DukeException {
```

Comment:
Overall, the `process` method seems to be longer than desired. Quoting our textbook:
> Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

Perhaps extract out some smaller methods?

Reply:
Alright thanks. Will do this as part of code quality improvements.